hibike/spawn_virtual_devices.py

```diff
@@ -0,0 +1,38 @@
+import csv
+import subprocess
+import shlex
+import re
+import os
+import atexit
+import time
+
+popens = []
+
+def get_virtual_ports():
+    socat = subprocess.Popen(shlex.split("socat -d -d pty,raw,echo=0 pty,raw,echo=0"), stderr=subprocess.PIPE)
+    popens.append(socat)
+    lines = [str(socat.stderr.readline()), str(socat.stderr.readline())]
+    #print(lines)
+    p1 = re.search(r'PTY is (/dev/pts/\d+)', lines[0]).group(1)
+    p2 = re.search(r'PTY is (/dev/pts/\d+)', lines[1]).group(1)
+    return p1, p2
+
+def spawn_device(device_type):
+    p1, p2 = get_virtual_ports()
+    fname = os.path.join(os.path.dirname(__file__), "virtual_device.py")
+    device = subprocess.Popen(shlex.split("python3.5 %s -d %s -p %s" % (fname, device_type, p1)))
```

Comment:
(copied from previous PR)
I think you should inserts some line breaks in here (maybe after some commas or parens). No need to use `\` or change the syntax, you can just use the already existing delimiters.